src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -286,26 +275,28 @@ class AbstractLQPNode : public std::enable_shared_from_this<AbstractLQPNode>, pr
    */
   std::optional<std::string> _table_alias;
 
-  /**
-   * For each column in the input node holds
-   *    - the ColumnID of this column in the output of this node
-   *    - INVALID_COLUMN_ID, if the column was created by this node.
-   *
-   * mutable, so it can be lazily initialized in output_column_ids_to_input_column_ids() overrides
-   */
-  mutable std::optional<std::vector<ColumnID>> _output_column_ids_to_input_column_ids;
+  // mutable, so it can be lazily initialized in output_column_origins() overrides
+  mutable std::optional<std::vector<LQPColumnOrigin>> _output_column_origins;
 
   /**
    * If named_column_reference.table_name is the alias set for this subtree, remove the table_name so that we
    * only operate on the column name. If an alias for this subtree is set, but this reference does not match
    * it, the reference cannot be resolved (see knows_table) and std::nullopt is returned.
    */
-  std::optional<NamedColumnReference> _resolve_local_alias(const NamedColumnReference& named_column_reference) const;
+  virtual std::optional<NamedColumnReference> _resolve_local_column_prefix(
+      const NamedColumnReference& named_column_reference) const;
+
+  /**
+   * @return the @param expression (you will probably want to pass in a deep_copy), with all ColumnOrigins pointing to
+   * their equivalent in a deep_copy()ed LQP
+   */
+  std::shared_ptr<LQPExpression> _adjust_expression_to_lqp(const std::shared_ptr<LQPExpression>& expression,
```

Comment:
It will make all ColumnOrigins in an Expression point to another LQP than they previously did. Ugly, but I think that's what we have to do with the current architecture.

Reply:
"adjust" is confusing, because it does not change anything. Also, isn't this a static method? I don't see `this` being used anywhere.

How about

```
std::shared_ptr<LQPExpression> _adapt_expression_to_different_lqp(const std::shared_ptr<LQPExpression>& expression, const std::shared_ptr<AbstractLQPNode>& other_lqp) const;
```

?

Reply:
Could be a free method - not being tied to a single type (but to two: Expression and LQPNode) - but I renamed it and made it static.